File: solutions/csharp/jedliks-toys/2/ElonsToys.cs
using System;

class RemoteControlCar
{
    private int METERS_PER_USE = 20;
    private int BATTERY_DRAIN_PER_USE = 1;
    private int batteryInPercent = 100;
    private int distanceDrivenInMeters = 0;

    public static RemoteControlCar Buy() => 
        new RemoteControlCar();

    public string DistanceDisplay() => 
        $"Driven {distanceDrivenInMeters} meters";

    public string BatteryDisplay() => 
        (batteryInPercent > 0) ? 
        $"Battery at {batteryInPercent}%" :
        "Battery empty";

    public void Drive()
    {
        if (batteryInPercent == 0) return; 
        batteryInPercent -= BATTERY_DRAIN_PER_USE;
        distanceDrivenInMeters += METERS_PER_USE;
    }
}

File: solutions/go/jedliks-toys/2/elons_toys.go
package elon

import "fmt"

// Drive drives car one time
func (car *Car) Drive() {
	if car.battery >= car.batteryDrain {
		car.distance += car.speed
		car.battery -= car.batteryDrain
	}
}

// DisplayDistance returns the distance driven so far as a formatted string
func (car Car) DisplayDistance() string {
	return fmt.Sprintf("Driven %d meters", car.distance)
}

// DisplayBattery returns the battery status as a formatted string
func (car Car) DisplayBattery() string {
	return fmt.Sprintf("Battery at %d%%", car.battery)
}

// CanFinish returns if the car can finish the track
func (car Car) CanFinish(trackDistance int) bool {
	return car.battery >= trackDistance/car.speed*car.batteryDrain
}
